### ISSUE TITLE
Desktop toolchain does not support Node 10 currently

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#5876]
We've had some reports of contributors being unable to build Desktop in Node 10, which has mostly been down to native node modules which weren't compatible with Node 10.

Some recent reports of users who encountered this: #5603, #5654, #5672

Note that this is a separate issue to how we build the Electron itself - we use [`app/.npmrc`](https://github.com/desktop/desktop/blob/master/app/.npmrc) to control how the native modules the app needs are built.

I believe https://github.com/desktop/desktop/pull/5644 is the last piece of the puzzle, but it's blocked on requiring clarification from another team about whether it's the right approach.

**Comment by shiftkey** [comment]
Friendly bump on this because Node 10 was just transitioned to LTS, meaning Node 8 support is going to be phased out - https://nodejs.org/en/blog/release/v10.13.0/

**Comment by outofambit** [comment]
for reference https://github.com/desktop/desktop/issues/5933#issuecomment-438130606

**Comment by shiftkey** [comment]
Tentatively pulling this into 1.6.1 because I have a working branch, but it needs some tooling to catch regressions. I plan to open a few PRs in the coming days in this order:

 - first, extract `legal-eagle` from the main `build.ts` into it's own module and wrap it in a `Promise`, so it can be chained better (top-level `await` isn't currently possible, as far as I'm aware)
 - next, introduce a regression test for any emitted CSS in the app bundle - if it contains a `$` this represents a SassScript value that wasn't encoded properly, so the app will run but _something_ won't look right
 - with all that in place, upgrade node-sass, catch all the new regressions, ensure it passes CI and test the app out
 - test out enabling CI to use Node 10 rather than Node 8. if that works, refresh the docs to simplify things

**Comment by shiftkey** [comment]
With #6661 just approved I'm going to move this to `1.7.0` to address the final stages of this PR:

 - review setup guides to drop mention of this constraint on Node 8
 - upgrade CI to Node 11 
 - PR in a change to documentation and build configs
 - confirm all builds work

**Comment by shiftkey** [comment]
I'm going to close this out as the underlying issue should be resolved for anyone using Node 10/11. 

Based on the feedback when reviewing #6661 I've captured the remaining work to explore Node 10/11 support in the project in these tasks: #6964 #6965
